internal/util/fscrypt.go

```diff
@@ -210,15 +217,49 @@ func initializeAndUnlock(ctx context.Context, fscryptContext *fscryptactions.Con
 	return nil
 }
 
+func getInodeEncryptedAttribute(p string) (bool, error) {
+	file, err := os.Open(p)
+	if err != nil {
+		return false, err
+	}
+	defer file.Close()
+
+	var attr int
+	_, _, errno := unix.Syscall(unix.SYS_IOCTL, file.Fd(), unix.FS_IOC_GETFLAGS,
+		uintptr(unsafe.Pointer(&attr)))
+	if errno != 0 {
+		return false, fmt.Errorf("error calling ioctl_iflags: %w", errno)
+	}
+
+	if attr&C.FS_ENCRYPT_FL != 0 {
+		return true, nil
+	}
+
+	return false, nil
+}
+
 // IsDirectoryUnlockedFscrypt checks if a directory is an unlocked fscrypted directory.
-func IsDirectoryUnlockedFscrypt(directoryPath string) error {
+func IsDirectoryUnlockedFscrypt(directoryPath, filesystem string) error {
 	if _, err := fscryptmetadata.GetPolicy(directoryPath); err != nil {
 		return fmt.Errorf("no fscrypt policy set on directory %q: %w", directoryPath, err)
 	}
 
-	_, err := xattr.Get(directoryPath, "ceph.fscrypt.auth")
-	if err != nil {
-		return fmt.Errorf("error reading ceph.fscrypt.auth xattr on %q: %w", directoryPath, err)
+	switch filesystem {
+	case "cephfs":
+		_, err := xattr.Get(directoryPath, "ceph.fscrypt.auth")
+		if err != nil {
+			return fmt.Errorf("error reading ceph.fscrypt.auth xattr on %q: %w", directoryPath, err)
+		}
+	case "ext4":
```

Comment:
maybe this should be the default case?

Reply:
Makes sense, but let me double check if other filesystems with fscrypt support set the inode encryption flag.

Reply:
Checked: ext4 and f2fs set the inode encrypted flag. Ceph FS doesn't support inode flags. Changing the ext4 to the default case.